### DISCUSSION TITLE
How to know which action can be used with `window.perform_action`?

### DISCUSSION THREAD

**Discussion by haphamdev** [wezterm/wezterm#6317] [Q&A] [answered]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

The doc of `window.perform_action` says:
> There are a number of actions that can be performed against a pane in a window when configured via the keys and mouse configuration options.

It seems to say that not all actions can be performed in that way.

I tried this 
```
wezterm.on("cev-tab-switching", function(win, pane)
  win.perform_action(act:ActivateTab(1), pane)
end)
```
and got an error
```
10:43:05.374 ERROR wezterm_gui::termwindow > while processing cev-tab-switching event: error converting Lua userdata to wezterm_dynamic::Value (error getting tostring: error converting Lua nil to function)
stack traceback:
        [C]: in method 'ActivateTab'
        /Users/hapham/.config/wezterm/keymappings.lua:163: in function 
```

Does it mean that `ActivateTab` cannot be used with `window.perform_action`?

**Comment by haphamdev** [comment]
Context: I was trying to configure a key mapping to jump between tabs with fuzzy search

I achieved it with `ShowLauncherArgs`
```
{
  key = "t",
  mods = "LEADER",
  action = act.ShowLauncherArgs({ flags = "FUZZY|TABS"}),
},
```
but still curious about `perform_action` which I think is really helpful.

**Comment by bew** [comment, marked as answer]
> `act:ActivateTab(1)`

This should be `act.ActivateTab(1)`
(otherwise it is trying to call it like `act.ActivateTab(act, 1)` which is wrong.)
